### PR TITLE
disttask: add Multitask test 

### LINKED ISSUE #46641: disttask: add Multitask test
## Enhancement
Disttask framework needs multi-task test.

### PR DESCRIPTION
**Pull request by JK1Zhang**
### What problem does this PR solve?

Issue Number: close #46641

Problem Summary: add Multitask test 

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: disttask/framework/framework_test.go [hunk 1]
```diff
@@ -399,3 +540,42 @@ func TestSchedulerDownManyNodes(t *testing.T) {
...
+
+	m := make([]sync.Map, num)
+	RegisterTaskMeta(&(m[0]), &testDispatcherExt{})
+	RegisterTaskMeta1(&(m[1]), &testDispatcherExt{})
+	RegisterTaskMeta2(&(m[2]), &testDispatcherExt{})
+
+	distContext := testkit.NewDistExecutionContext(t, 1)
+	dispatcher.MockOwnerChange = func() {
+		distContext.SetOwner(0)
+	}
```
Comment (ywqzzy) [inline disttask/framework/framework_test.go:557]: Useless code.

File: disttask/framework/framework_test.go [hunk 2]
```diff
@@ -399,3 +540,42 @@ func TestSchedulerDownManyNodes(t *testing.T) {
...
+	defer dispatcher.ClearDispatcherFactory()
+	defer scheduler.ClearSchedulers()
+	num := 3
+
+	m := make([]sync.Map, num)
+	RegisterTaskMeta(&(m[0]), &testDispatcherExt{})
+	RegisterTaskMeta1(&(m[1]), &testDispatcherExt{})
+	RegisterTaskMeta2(&(m[2]), &testDispatcherExt{})
+
+	distContext := testkit.NewDistExecutionContext(t, 1)
```
Comment (ywqzzy) [inline disttask/framework/framework_test.go:554]:
```suggestion
	distContext := testkit.NewDistExecutionContext(t, 3)
```
More nodes will be better

File: disttask/framework/framework_test.go [hunk 3]
```diff
@@ -210,6 +291,66 @@ func DispatchTaskAndCheckState(taskKey string, t *testing.T, m *sync.Map, state
...
+				tasks[0] = task
+				require.NoError(t, err)
+				require.NotNil(t, task)
+				if task.State != proto.TaskStatePending && task.State != proto.TaskStateRunning && task.State != proto.TaskStateCancelling && task.State != proto.TaskStateReverting {
+					break
+				}
+			}
+			require.NoError(t, failpoint.Disable("github.com/pingcap/tidb/disttask/framework/scheduler/MockExecutorRunErr"))
+		} else {
+			taskID[i], err = mgr.AddNewGlobalTask("key"+fmt.Sprintf("%d", i), proto.Int2Type(i+2), 8, nil)
```
Comment (ywqzzy) [inline disttask/framework/framework_test.go:326]:
```suggestion
			taskID[i], err = mgr.AddNewGlobalTask(fmt.Sprintf("key%d", i), proto.Int2Type(i+2), 8, nil)
```

File: disttask/framework/framework_test.go [hunk 4]
```diff
@@ -152,6 +189,50 @@ func RegisterTaskMeta(m *sync.Map, dispatcherHandle dispatcher.Extension) {
 	})
 }
 
+func RegisterTaskMeta1(m *sync.Map, dispatcherHandle dispatcher.Extension) {
```
Comment (ywqzzy) [inline disttask/framework/framework_test.go:192]: We can rename the function

**Comment by JK1Zhang** [comment]
\retest

**Review by ywqzzy — APPROVED** [review]
LGTM

**Review by tangenta — APPROVED** [review]

**Pull request merged**